astropy/table/column.py

```diff
@@ -383,6 +383,13 @@ def __new__(cls, data=None, name=None,
             if meta is None:
                 meta = data.info.meta
 
+        elif isinstance(data, list) and data == []:
+            # We should special case the empty list since np.array([], dtype)
+            # does not return an empty array with the requested dtype, and in
+            # numpy 1.20-dev this is giving a DeprecationWarning.
+            # TODO: Adjust when/if the numpy behaviour behaviour changes.
+            # Hopefully, this clause can again be removed.  NUMPY_LT_1_20.
+            self_data = np.empty(0, dtype=dtype)
```

Comment:
One other solution that is possible if you know that the list represents effectively a 1-D dataset is to use:
`np.empty(len(data), dtype=dtype)` and then assign to that result.  That does have slightly different semantics though.

Reply:
Yes, that might indeed work. But I think the standard case should just go through `np.array(...)`, since `Column` is an ndarray subclass.

The whole thing does make one wonder whether the new behaviour in numpy is in fact an improvement, since it doesn't seem to move to the "more logical" state - re-opening the numpy issue.